From 14bd7dd6a896574333954c09e4f9774694238ab3 Mon Sep 17 00:00:00 2001
From: Zeyu Fan <Zeyu.Fan@amd.com>
Date: Fri, 27 May 2016 14:13:49 -0400
Subject: [PATCH 0506/1722] drm/amd/dal: Disabling HDMI and DVI clock sharing.

Modified is_sharable_clk_src function to
make HDMI and DVI signal not clock sharable, otherwise can
cause screen not lightening up.

Also renamed function resource_are_streams_clk_sharable to
resource_are_streams_timing_synchronizable

Signed-off-by: Zeyu Fan <Zeyu.Fan@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c          |  2 +-
 drivers/gpu/drm/amd/dal/dc/core/dc_resource.c | 15 +++++++++++----
 drivers/gpu/drm/amd/dal/dc/inc/resource.h     |  2 +-
 3 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index 77a93f5..ae9d4d0 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -544,7 +544,7 @@ static void program_timing_sync(
 			if (!ctx->res_ctx.pipe_ctx[j].stream)
 				continue;
 
-			if (resource_are_streams_clk_sharable(
+			if (resource_are_streams_timing_synchronizable(
 					ctx->res_ctx.pipe_ctx[j].stream,
 					ctx->res_ctx.pipe_ctx[i].stream)) {
 				tg_set[group_size] =
diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c b/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c
index 809f6f16..ecd5f82 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c
@@ -160,11 +160,10 @@ void resource_reference_clock_source(
 		res_ctx->dp_clock_source_ref_count++;
 }
 
-bool resource_are_streams_clk_sharable(
+bool resource_are_streams_timing_synchronizable(
 	const struct core_stream *stream1,
 	const struct core_stream *stream2)
 {
-
 	if (stream1->public.timing.h_total != stream2->public.timing.h_total)
 		return false;
 
@@ -201,7 +200,15 @@ static bool is_sharable_clk_src(
 	if (dc_is_dp_signal(pipe_with_clk_src->stream->signal))
 		return false;
 
-	if (!resource_are_streams_clk_sharable(
+	if (dc_is_hdmi_signal(pipe_with_clk_src->stream->signal)
+			&& dc_is_dvi_signal(pipe->stream->signal))
+		return false;
+
+	if (dc_is_hdmi_signal(pipe->stream->signal)
+			&& dc_is_dvi_signal(pipe_with_clk_src->stream->signal))
+		return false;
+
+	if (!resource_are_streams_timing_synchronizable(
 			pipe_with_clk_src->stream, pipe->stream))
 		return false;
 
@@ -754,7 +761,7 @@ static struct core_stream *find_pll_sharable_stream(
 				DC_STREAM_TO_CORE(target->public.streams[j]);
 
 			/* We are looking for non dp, non virtual stream */
-			if (resource_are_streams_clk_sharable(
+			if (resource_are_streams_timing_synchronizable(
 						stream_needs_pll, stream_has_pll)
 				&& !dc_is_dp_signal(stream_has_pll->signal)
 				&& stream_has_pll->sink->link->public.connector_signal
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/resource.h b/drivers/gpu/drm/amd/dal/dc/inc/resource.h
index ad6a196..0147f41 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/resource.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/resource.h
@@ -65,7 +65,7 @@ void resource_reference_clock_source(
 		struct resource_context *res_ctx,
 		struct clock_source *clock_source);
 
-bool resource_are_streams_clk_sharable(
+bool resource_are_streams_timing_synchronizable(
 		const struct core_stream *stream1,
 		const struct core_stream *stream2);
 
-- 
2.7.4

